URAP-charter/web_scraping
# 2 11 2021 george fix mongo

The purpose of this branch is to debug the MongoDB pipeline and get our scraper to communicate with it. 

Prior to this branch, it was observed that the Mongo Client would connect and immediately close, and sometimes it would simply not connect to the database. Fixes made were:
1. Adding the "items" to the mongodb setup in pipelines.from_crawler
2. Change the Mongo URI in settings.py to include the port
3. Resolve tsv vs csv issues in the input object (not in this branch to avoid merge conflicts with @pmasaya )

To Test this PR:
1. Start up MongoDB. I recommend using Docker, as explained in the README
2. Set up your settings.py as mentioned in the README -- basically just ensure the "mongo pipeline", "MONGO_URI", and "MONGO_DATABASE" lines are all uncommented
3. Run the spider as explained in the README -- ensure that you do NOT have the "-o" flag and argument, because Mongo will be handling the output
4. You should notice that files are being written to Mongo
5. After the spider is done, you can check your Mongo container:

- docker exec -it mongodb bash
- mongo
- show dbs;
You should notice that the "schoolSpider" db has some data! If you're more familiar with Mongo, you are welcome to explore that.

Comment below if you run into any issues.
